README.md

```diff
@@ -1 +1,826 @@
-# message-tracing-with-stream-processor
\ No newline at end of file
+[![Build Status](https://travis-ci.org/ballerina-guides/message-tracing-with-stream-processor.svg?branch=master)](https://travis-ci.org/ballerina-guides/message-tracing-with-stream-processor)
+
+# Integration with WSO2 Stream Processor
+
+WSO2 stream processor provides us with distributed message tracing. The Distributed Message Tracer allows you to trace the events which are produced while ballerina services serve for requests. The ballerina services send these tracing data as WSO2 events.
+        
+  >This guide provides instructions on how Ballerina can be used to integrate with Stream-Processor Message Tracer.
+  
+The following are the sections available in this guide.
+
+- [What you'll build](#what-you’ll-build)
+- [Prerequisites](#prerequisites)
+- [Implementation](#implementation)
+- [Testing](#testing)
+- [Testing with distributed message tracing](#testing-with-distributed-message-tracer.)
+     - [Traces](#views-of-traces)
+
+## What you’ll build
+To perform this integration with Distributed Message Tracer, a real world use case of a very simple student management system is used.
+This system will illustrate the manipulation of student details in a school/college management system. The administrator will be able to perform the following actions in this service.
+
+    - Add a student's details to the system.
+    - List down all the student's details who are registered in the system.
+    - Delete a student's details from the system by providing student ID.
+    - Generate a mock error (for observability purposes).
+    - Get a student's marks list by providing student ID.
+
+![Honeycomb](images/ballerina-sp.svg "Ballerina-Honeycomb")
+
+- **Make Requests** : To perform actions on student  management service, a console based client program has been written in Ballerina for your ease of making requests.
+
+## Prerequisites
+ 
+- [Ballerina Distribution](https://ballerina.io/learn/getting-started/)
+- [MySQL](https://github.com/ballerina-guides/message-tracing-with-stream-processor/blob/master/resources/testdb.sql)
+- [WSO2 - Stream Processor v4.3.0 or above](https://github.com/wso2/product-sp/releases)
+- A Text Editor or an IDE 
+
+## Implementation
+
+> If you want to skip the basics, you can download the GitHub repo and continue from the [Testing](#testing) section.
+
+### Implementing database
+ - Start MySQL server in your local machine.
+ - Create a database with name `testdb` in your MySQL localhost. If you want to skip the database implementation, then directly import the [testdb.sql](https://github.com/ballerina-guides/message-tracing-with-stream-processor/blob/master/resources/testdb.sql) file into your localhost. You can find it in the Github repo.
```

Comment:
Change to "Create a database named..."

Change "Github" to "GitHub". Apply elsewhere if applicable,